tensorflow_addons/image/color_ops.py

```diff
@@ -27,49 +27,41 @@
 from functools import partial
 
 
-def equalize_image(image: TensorLike) -> tf.Tensor:
-    """Implements Equalize function from PIL using TF ops."""
-
-    @tf.function
-    def scale_channel(image, channel):
-        """Scale the data in the channel to implement equalize."""
-        image_dtype = image.dtype
-        image = tf.cast(image[:, :, channel], tf.int32)
-
-        # Compute the histogram of the image channel.
-        histo = tf.histogram_fixed_width(image, [0, 255], nbins=256)
-
-        # For the purposes of computing the step, filter out the nonzeros.
-        nonzero = tf.where(tf.not_equal(histo, 0))
-        nonzero_histo = tf.reshape(tf.gather(histo, nonzero), [-1])
-        step = (tf.reduce_sum(nonzero_histo) - nonzero_histo[-1]) // 255
-
-        def build_lut(histo, step):
-            # Compute the cumulative sum, shifting by step // 2
-            # and then normalization by step.
-            lut = (tf.cumsum(histo) + (step // 2)) // step
-            # Shift lut, prepending with 0.
-            lut = tf.concat([[0], lut[:-1]], 0)
-            # Clip the counts to be in range.  This is done
-            # in the C code for image.point.
-            return tf.clip_by_value(lut, 0, 255)
-
-        # If step is zero, return the original image.  Otherwise, build
-        # lut from the full histogram and step and then index from it.
-
-        if step == 0:
-            result = image
-        else:
-            result = tf.gather(build_lut(histo, step), image)
+def scale_channel(image: TensorLike, channel: int) -> tf.Tensor:
```

Comment:
Let's prefix the internal function with `_` so change `scale_channel` to `_scale_channel`.